components/webapps/blacklist-whitelist-service/src/main/java/com/wso2telco/services/bw/Queries.java

```diff
@@ -56,360 +51,262 @@
+						try {
+							blackListWhiteListService.getBlacklistAsZip(dto, outputStream);
+						} catch (Exception e) {
+							throw new WebApplicationException(e);
+						}
 
-			try {
-				blackListWhiteListService.removeBlacklist(apiId, msisdn);
-
-				StringBuilder jsonreturn = new StringBuilder();
-				jsonreturn.append("{").append("\"Success\":").append("{").append("\"messageId\":\"")
-						.append("Remove Number").append("\",").append("\"text\":\"");
-				jsonreturn.append("Blacklist number successfully removed ").append("\",").append("\"variables\":\"")
-						.append(msisdn).append("\"").append("}}");
-				return Response.status(Response.Status.OK).entity(jsonreturn.toString()).build();
-			} catch (BusinessException e) {
-				LOG.error("", e);
 
-				return Response.status(Response.Status.BAD_REQUEST).entity(e.getErrorType()).build();
-			}
-		} else {
-			return Response.status(Response.Status.BAD_REQUEST).entity(erroMsg.toString()).build();
-		}
+					}
+				};
 
+				return Response.status(Response.Status.OK).entity(stream).build();
+			} catch (Exception e) {
+				LOG.error("Error building zip file",e);
+				return Response.status(Response.Status.BAD_REQUEST).entity(e.getMessage()).build();
+			}
 	}
 
 	/**
-	 * GET method for creating an instance of QueriesResource
-	 *
-	 * @param address,requestedAccuracy
-	 *            representation for the new resource
-	 * @return an HTTP response with content of the created resource
+	 * API used for blacklisting/whitelisting a list of MSISDNs
+	 * @param uploadedInputStream text file with records separated by a new line
+	 * @param api API ID for this list of records
+	 * @param app APP ID for this list of records
+	 * @param action Action (i.e: blacklist/whitelist)
+	 * @param user userId (login user i.e: admin's user name)
+	 * @return
 	 */
-	@Deprecated
 	@POST
-	@Path("/whitelist/{MSISDN}")
-	@Consumes("application/json")
-	@Produces("application/json")
-	public Response provisionWhiteListedNumber(@PathParam("MSISDN") String msisdn, String jsonBody) {
-
-		LOG.debug("provisionWhiteListedNumber Triggerd  jsonBody :" + jsonBody + " , msisdn :" + msisdn);
-
-		Gson gson = new GsonBuilder().serializeNulls().create();
-
-		WhiteList whiteListReq = gson.fromJson(jsonBody, WhiteList.class);
-
-		String subscriptionID = whiteListReq.getSubscriptionID();
-		String apiID = whiteListReq.getAPIID();
-		String applicationID = whiteListReq.getApplicationID();
-		final StringBuilder errorMSG = new StringBuilder();
-
-		errorMSG.append("{").append("\"Failed\":".intern()).append("{").append("\"messageId\":\"".intern())
-				.append("Whitelisted Number".intern()).append("\",");
-		errorMSG.append("\"text\":\"").append("Whitelisted Number could not be added to the system ".intern())
-				.append("\",");
-		errorMSG.append("\"variables\":\"").append(msisdn).append("\"").append("}}");
-
-		if (subscriptionID != null && apiID != null && applicationID != null) {
+	@Path("/bulkAdd/{app}/{api}")
```

Comment:
will avoid using camel cases in endpoint URLs (downlbulkAddoadAsZip) and replace with hyphen character separated